src/stores/simulations.ts

```diff
@@ -62,6 +63,9 @@ export const useSimulationStore = defineStore('simulation', () => {
       setCurrentStep(availableSteps[indexPreviousStepNotFinal].step)
       setCurrentSubStep(availableSteps[indexPreviousStepNotFinal].subStep)
     }
+    if (currentStep.value == 2 && currentSubStep.value == 3) {
```

Comment:
I am not sure to understand this code 
Here, if step = 2 and substep = 3 , we call again goToPreviousStep. So , we never go to step = 2 and substep = 3 ?
In what condition do we arrive in current = 2 and sub-step = 3?

Reply:
Step 2.3 is not completely part of the flow. When you have positioned the obstacles on the roof in step 2.1, you go to 2.2 and you want to move on in the simulation. There are two possibilities : either the surface is not favorable after deduction of the obstacles, then you pass in 2.3 and it is a dead end of the flow ; either the surface is favorable and therefore you go to 3.1. That's why when everything is fine, we go from 2.2 to 3.1 ans skip 2.3.